From 925655d580bafe2ce1cb3af60676bc3693393782 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.jz@renesas.com>
Date: Mon, 19 Apr 2021 17:23:07 +0700
Subject: [PATCH 179/180] arm64: dts: rzg2-smarc: enable DMA for RSPI1

Enable DMA for RSPI1 TX/RX on RZ/G2L development board

Signed-off-by: Binh Nguyen <binh.nguyen.jz@renesas.com>
---
 arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi b/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
index 88deedd..f14c35a 100644
--- a/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
@@ -562,6 +562,8 @@
 #define RZADMA_SSI0_RX	2
 #define RZADMA_SSI1_TX	3
 #define RZADMA_SSI1_RX	4
+#define RZADMA_RSPI1_TX	5
+#define RZADMA_RSPI1_RX	6
 
 /* DMA Configuration */
 &dma0  {
@@ -569,12 +571,16 @@
 	slave_id = <RZADMA_SSI0_TX
 		    RZADMA_SSI0_RX
 		    RZADMA_SSI1_TX
-		    RZADMA_SSI1_RX>;
+		    RZADMA_SSI1_RX
+		    RZADMA_RSPI1_TX
+		    RZADMA_RSPI1_RX>;
 	/* Address of data source/destination */
 	addr = <0x10049C18	/* SSIFTDR_0 */
 		0x10049C1C	/* SSIFRDR_0 */
 		0x1004A018	/* SSIFTDR_1 */
 		0x1004A01C	/* SSIFRDR_1 */
+		0x1004B004	/* SPDR_1 */
+		0x1004B004	/* SPDR_1 */
 		>;
 	/* Channel Configuration Register (CHCFG) for slave device */
 	/* .reqd .loen .hien .lvl .am .sds .dds .tm */
@@ -582,6 +588,8 @@
 		 0x0 0x0 0x1 0x0 0x2 0x1 0x1 0x0	/* SSI ch0 RX (req sig=SSIFRDR_0) */
 		 0x1 0x0 0x1 0x0 0x2 0x1 0x1 0x0	/* SSI ch1 TX (req sig=SSIFTDR_0) */
 		 0x0 0x0 0x1 0x0 0x2 0x1 0x1 0x0	/* SSI ch1 RX (req sig=SSIFRDR_0) */
+		 0x1 0x0 0x1 0x1 0x2 0x0 0x0 0x0	/* RSPI1 TX (SPDR_1) */
+		 0x0 0x0 0x1 0x1 0x2 0x0 0x0 0x0	/* RSPI1 RX (SPDR_1) */
 		 >;
 	/* DMARS for slave device */
 	/* .rid .mid */
@@ -589,6 +597,8 @@
 		 0x2 0x95	/* SSI ch0 RX */
 		 0x1 0x96	/* SSI ch1 TX */
 		 0x2 0x96	/* SSI ch1 RX */
+		 0x1 0xA6	/* RSPI ch1 TX */
+		 0x2 0xA6	/* RSPI ch1 RX */
 		 >;
 };
 
@@ -676,6 +686,9 @@
 	pinctrl-0 = <&spi1_pins>;
 	pinctrl-names = "default";
 	status = "okay";
+	dmas = <&dma0 RZADMA_RSPI1_TX>,
+	       <&dma0 RZADMA_RSPI1_RX>;
+	dma-names = "tx", "rx";
 };
 
 #if (GPT_SUPPORT)
-- 
2.7.4

